### PR TITLE
Cleanup unsupported python code (except for adodbapi)

### PR DESCRIPTION
**Pull request by Avasam**
This is ~~the second~~ *a* PR in a series concerning code cleanup. With the final goal to make basic type-checking validation of public methods possible, easing the addition of 3.7+ type annotations.

This cleans-up most unreachable code. Mostly due to unsupported python versions.
- Update obsolete Python1 comments
- Remove Python2 specific code
  - Update Python3/Python2 specific comments
  - Added a `TODO` to the comments that mentioned python 2/3 specific code. But that I didn't know how to address.
  - ~~Replaced use of deprecated Types~~ (done in multiple PRs listed below)
  - ~~Obsolete and redundant aliases~~ #2087
  - ~~Redundant object subclassing~~ #2086
  - ~~Obsolete references to older python unicode compatibility~~ #2085
- Update platform/version checks
- Removed unreachable code
- ~~Removed duplicate methods found~~ #2050
- ~~Used bytes-string (`b""`) directly instead of `"".encode()` when applicable~~ #2046
- ~~Use of raw strings when pattern-matching~~ #2045

Note these changes are for pure python code only. It'd be nice to do the same for c-modules, but I don't have the necessary knowledge or expertise there. I'm sure there's more I could do, if I missed anything, please let me know!

### CODE REVIEW

**Comment by vernondcole** [comment]
We have a problem here.
The refactoring of the adodbapi code is extensive. It has needed to be done for a long time, and I have neglected it, so I am thankful for the work. Don't get that wrong.
The problem is: we cannot test it.
Microsoft Azure has recently turned off the Sql Server machine which I have kept running for several years to use for testing. I don't think it has been used anyway, and was a poor arrangement to start with, so it's definitely time for something better.

1) DO NOT MERGE this PR until we can run a full test of adodbapi.

2) How can we do that?  

Is there some facility local to the CI (or nearby) on which we can maintain, or spin up, a small SQL Server instance?

**Comment by vernondcole** [comment]
I have added a comment on GitHub, and added myself as a reviewer. I am also
commenting here on email to make sure that everyone gets this.

DO NOT MERGE THIS UNTIL ADODBAPI HAS BEEN FULLY TESTED!

It will probably pass, and I hope it does, but it must be done. This is
great work, and we cannot spoil it by skipping an essential step which has
been too long ignored.

**Comment by Avasam** [comment]
@vernondcole Thanks for your input! If there are currently untestable areas that would require a lot of work to setup. I can always split up those sections in a different PR so that it's tackled separately. Or let you test all at once since it'll have to be done at some point anyway.

**Comment by vernondcole** [comment]
If we can come up with an expeditious testing method, I don't see a need
for a separate PR. On the other hand, if testing slows down a release
substantially, then it would be a good idea. Don't split it yet, but let's
keep the idea in mind.

I could certainly write a Salt script, for my local development
environment, to spin up three virtual machines to run the database servers
and another to build and test pywin32. But it would be a better solution to
do that in the CI environment, I suspect.

We may want to set up two levels of tests, a quirk one for automatic
go/no-go in the CI, and a more detailed one run on a periodic basis.
Testing adodbapi takes several minutes at best even when the database
servers are kept running. I don't know what our CI system is capable of,
and if someone who does wanted to start a separate conversation about this
subject, I would love it.

File: Pythonwin/Scintilla/src/LexGen.py
Comment (mhammond) [inline Pythonwin/Scintilla/src/LexGen.py:7]: scintilla is "vendored" but I don't think yet "forked" - so we should probably skip mods there.

**Review by mhammond — COMMENTED** [review]
Until #1986 lands it will be tricky to look closely, but a very quick look seems great.

I think we should split adodbapi, but the question is, where: @vernondcole in #1986 there are a couple of commits which are tool generated, first of which sorts imports. How do you feel about an untested change which just does this, for example? We could split things at one of the "tool generated" steps which might make things easier - eg, I tend to let black reformat for me without bothering to re-test what it did.

File: Pythonwin/Scintilla/src/LexGen.py
Comment (Avasam) [reply Pythonwin/Scintilla/src/LexGen.py:7]: By "mods" you mean "modifications" right?
Make sense if it's vendored.

I'm not certain exactly what you mean by "forked" (I'm lacking a bit of context here). As in it will come to live in pywin32 from somewhere else? Or as in it will be taken out of pywin32 to exist on its own?

> Until https://github.com/mhammond/pywin32/pull/1986 lands it will be tricky to look closely, but a very quick look seems great.

I'm in no hurry, let's do things one step at a time with isolated changes :)

Comment (Avasam) [reply Pythonwin/Scintilla/src/LexGen.py:7]: Btw if part of this PR is blocked because it touches vendored code. I can simply split off those changes.

File: Pythonwin/pywin/idle/PyParse.py
Comment (Avasam) [inline Pythonwin/pywin/idle/PyParse.py:128]: How should this be updated to remove python2 support code? Is it just about removing `_tran = "".join(_tran)` ?

File: Pythonwin/pywin/mfc/dialog.py
Comment (Avasam) [inline Pythonwin/pywin/mfc/dialog.py:96]: How should this be updated to remove old python support code?

File: setup.py
Comment (Avasam) [inline setup.py:883]: Which part exactly can go now?

File: adodbapi/remote/server.py
Comment (Avasam) [inline adodbapi/remote/server.py:95]: What's the change needed here ?

File: com/win32com/client/build.py
Comment (Avasam) [inline com/win32com/client/build.py:634]: What strides are we talking about? I can only find references to `memoryview.strides` and `numpy.ndarray.strides`. Is this comment still relevant?

File: com/win32comext/axscript/test/testHost.py
Comment (Avasam) [inline com/win32comext/axscript/test/testHost.py:140]: Not entirely certain if it's the test or just the message that should be updated here.

File: com/win32com/client/genpy.py
Comment (Avasam) [inline com/win32com/client/genpy.py:41]:
```suggestion
    pythoncom.VT_BSTR: "str",
```

**Comment by Avasam** [comment]
Refactoring of adodbapi code has been completely split off to #2094 so that this here PR can be testable and reviewable sooner.

File: Pythonwin/Scintilla/src/LexGen.py
Comment (Avasam) [reply Pythonwin/Scintilla/src/LexGen.py:7]:
```suggestion
# Requires Python 2.4 or later
```
I'll revert this for now. It's only a comment change anyway

File: Pythonwin/pywin/idle/readme.txt
Comment (mhammond) [inline Pythonwin/pywin/idle/readme.txt:9]: I think this should be reworded to reflect something like "if the files in this directory are newer than the ones shipped with your Python version" or similar.

File: Pythonwin/Scintilla/src/LexGen.py
Comment (mhammond) [inline Pythonwin/Scintilla/src/LexGen.py:204]: this code is "vendored", so I don't really want to fork it. I don't think it's used in practice by pywin32.

File: Pythonwin/pywin/framework/app.py
Comment (mhammond) [inline Pythonwin/pywin/framework/app.py:434]: This PR doesn't look like it was auto-generated, so why this change?

File: Pythonwin/pywin/mfc/dialog.py
Comment (mhammond) [reply Pythonwin/pywin/mfc/dialog.py:96]: I really can't recall

File: com/win32com/client/build.py
Comment (mhammond) [reply com/win32com/client/build.py:634]: No, I think this comment can be killed.

File: com/win32com/client/makepy.py
Comment (mhammond) [inline com/win32com/client/makepy.py:428]: unrelated to this PR, but I suspect this is wrong now - when this file is being read I believe it will be assumed utf-8. Ditto the `mbcs` reference in build.py

File: com/win32comext/axscript/test/testHost.py
Comment (mhammond) [reply com/win32comext/axscript/test/testHost.py:140]: I think killing the comment and references to `PyScript_Exc` seems fine.

File: com/win32comext/taskscheduler/test/test_addtask_1.py
Comment (mhammond) [inline com/win32comext/taskscheduler/test/test_addtask_1.py:36]: maybe "py.exe" with no path makes sense here and below, and omit the `SetWorkingDirectory` call?

File: setup.py
Comment (mhammond) [reply setup.py:817]: I don't think it is worth it frankly.

File: win32/Lib/winnt.py
Comment (mhammond) [inline win32/Lib/winnt.py:183]: why this change?

**Review by mhammond — COMMENTED** [review]
Thanks for persevering!

File: Pythonwin/pywin/framework/app.py
Comment (Avasam) [reply Pythonwin/pywin/framework/app.py:434]: Pretty sure that's a leftover from cleaning up / reorganizing imports. I don't particularly care whether the comment is same-line or above (as long as it's not too long, and it is under 80 chars). I can revert this if you wanna reduce changes.

File: Pythonwin/Scintilla/src/LexGen.py
Comment (Avasam) [reply Pythonwin/Scintilla/src/LexGen.py:204]: I meant to revert all scintilla-related code (see your other similar comment about Scintilla). Missed this one, thanks!

File: Pythonwin/Scintilla/src/LexGen.py
Comment (Avasam) [inline Pythonwin/Scintilla/src/LexGen.py:206]:
```suggestion
    return cmp(a.lower(), b.lower())
```

File: Pythonwin/pywin/idle/readme.txt
Comment (Avasam) [reply Pythonwin/pywin/idle/readme.txt:9]: I'm not sure how I would integrate that text. If they are newer, then what?

The current text seems to imply that this directory was based on IDLE from a specific python version (>3.7). What I'm not sure about, is whether saying that Python 3.8 is the version it's based on is accurate. Or if I'm even understanding this message correctly.

Also by "later" does this text mean "newer" ?

File: Pythonwin/pywin/idle/PyParse.py
Comment (Avasam) [reply Pythonwin/pywin/idle/PyParse.py:153]: I wanna say the parenthesis and the fact that IDLE prints some text on opening is what's important to this comment, not the version number (see image, it would seem it's still accurate).
image

But if you see value in keeping a specific version number in there to know for which Python version this code and comment was initially written, then I can leave it in.

File: adodbapi/remote/server.py
Comment (Avasam) [reply adodbapi/remote/server.py]: Moved adodbapi changes: https://github.com/mhammond/pywin32/pull/2094#discussion_r1285238410

File: com/win32com/client/makepy.py
Comment (Avasam) [reply com/win32com/client/makepy.py:428]: Seems like a simple enough change, `encoding="mbcs"` in two files. Can be done in a separate PR.

File: com/win32comext/axscript/test/testHost.py
Comment (Avasam) [reply com/win32comext/axscript/test/testHost.py:140]: Do you mean that removing references to `PyScript_Exc` (and by extension, removing the `testPythonExceptions` test) is fine, or did you mean that having the references to `PyScript_Exc` is fine?

Left it for now, updated comment to mention Python 2 in the past tense.

File: com/win32comext/taskscheduler/test/test_addtask_1.py
Comment (Avasam) [reply com/win32comext/taskscheduler/test/test_addtask_1.py:36]: If the tests are still valid, sounds good to me.

File: win32/Lib/winnt.py
Comment (Avasam) [reply win32/Lib/winnt.py:183]: Bad merge, it should be as was in this PR: https://github.com/mhammond/pywin32/pull/2050/files#diff-929c99ca621bd62184b36b3c6192c02547a50e4bc9e45a1968a2565622f454d2

File: Pythonwin/pywin/framework/app.py
Comment (Avasam) [reply Pythonwin/pywin/framework/app.py:422]: `raw_input` doesn't exist on python 3. So 
```py
import code

sys.modules["builtins"].input = Win32RawInput
```
was the only code that can run.

`Win32Input` is no longer used.

The concept of raw input vs regular input doesn't exist in python 3 (afaik). So the name didn't make sense anymore. So I removed the old unused `Win32Input` and renamed `Win32RawInput` to `Win32Input`

File: Pythonwin/pywin/idle/readme.txt
Comment (mhammond) [reply Pythonwin/pywin/idle/readme.txt:9]: This comment was trying to reflect that the files in this dir also exist in IDLE, but may have been released as part of pywin32 before a Python release was made. So yeah, I guess this entire readme can now just say something like "these files have been forked from IDLE" or similar.

File: Pythonwin/pywin/framework/app.py
Comment (mhammond) [reply Pythonwin/pywin/framework/app.py:422]: yeah, thanks, I understood the intent - I was just wondering if you actually checked that, eg, `input("enter something:")` from the interactive window still actually worked? Sadly I have no windows machine handy at the moment.

File: Pythonwin/pywin/framework/app.py
Comment (mhammond) [reply Pythonwin/pywin/framework/app.py]: I'm fine with this isolated change but in general would prefer to avoid "personal subjective" stylistic changes (whereas "automated subjective" stylistic changes by commonly used tools are a necessary evil :)

File: Pythonwin/pywin/idle/PyParse.py
Comment (mhammond) [reply Pythonwin/pywin/idle/PyParse.py:153]: oops, yes, I agree

File: com/win32comext/axscript/test/testHost.py
Comment (mhammond) [reply com/win32comext/axscript/test/testHost.py]: I think removing `PyScript_Exc` and that test is probably best.

File: com/win32comext/taskscheduler/test/test_addtask_1.py
Comment (mhammond) [reply com/win32comext/taskscheduler/test/test_addtask_1.py]: I'd not be at all surprised if these went stale somewhere since python 2.3, but the rest of the test infrastructure tends to use `py.exe` so yeah, let's do that here and we can worry about whether it actually still works later.

File: win32/Lib/winnt.py
Comment (mhammond) [reply win32/Lib/winnt.py]: yeah - I did mean to say that these should just be removed - that PR removed duplicates of this odd definition, can this PR just remove them completely?

File: Pythonwin/pywin/framework/app.py
Comment (Avasam) [reply Pythonwin/pywin/framework/app.py]: Yeah I'd rather leave this in whatever state won't cause a change for this PR.

File: Pythonwin/pywin/idle/readme.txt
Comment (Avasam) [reply Pythonwin/pywin/idle/readme.txt:9]: Ok. I have pushed a wording change suggestion that should not fall out of date.

File: win32/Lib/winnt.py
Comment (Avasam) [inline win32/Lib/winnt.py:182]: Removed as per requested

File: win32/Lib/winnt.py
Comment (Avasam) [inline win32/Lib/winnt.py:175]: `(WORD)` and `(DWORD)` seem like invalid leftovers from C casting transpilation.

`SORTIDFROMLCID` and `LANGIDFROMLCID` are not referenced anywhere.

File: win32/Lib/winnt.py
Comment (Avasam) [inline win32/Lib/winnt.py:168]: `(WORD)` and `(DWORD)` seem like invalid leftovers from C casting transpilation.

Same as `win32con.py`

File: setup.py
Comment (Avasam) [reply setup.py:882]: Ah, this is related to the same "temp dir" comment as above.

File: com/win32comext/taskscheduler/test/test_addtask_2.py
Comment (mhammond) [inline com/win32comext/taskscheduler/test/test_addtask_2.py:23]: we should really do the same as we did for test_addtask_1.py here - do you mind opening a followup (either an issue or PR)? Not I'm going to be largely offline for 2 weeks.

**Review by mhammond — APPROVED** [review]
Thanks!

File: com/win32comext/taskscheduler/test/test_addtask_2.py
Comment (Avasam) [reply com/win32comext/taskscheduler/test/test_addtask_2.py:23]: https://github.com/mhammond/pywin32/pull/2099

**Pull request merged**
